rootfs/ingest/python/qserv/jsonparser.py

```diff
@@ -290,7 +292,7 @@ def raise_error(responseJson: dict, retry_attempts: int = -1, max_retry_attempts
         retrying request. False if no error in JSON response.
 
     """
-    if retry_attempts < max_retry_attempts:
+    if retry_attempts != -1 and retry_attempts < max_retry_attempts:
```

Comment:
Seeing the negative number of `retry_attempts` is confusing. I must admit I don't quite get what would be consequences in a respect of triggering more retries in a case when this method is called with default values of the last two arguments:
```
raise_error(responseJson)
```
It looks like, any error will trigger a retry regardless of the actual number of retries or ,limits on the number of such retries?

Reply:
`raise_error(responseJson)` does not perform any retry at all. Retry feature in only used once, when monitoring async queries... I should make this code simpler and more understandable!